homeassistant/util/logging.py

```diff
@@ -53,9 +59,15 @@ def handle(self, record: logging.LogRecord) -> None:
             self._reset_counters(record.created)
 
         module_name = record.name
-        if module_name == __name__ or module_name in self._warned_modules:
+
+        skip_flag: bool | None = self._module_log_count_skip_flags.get(module_name)
+        if skip_flag:
             return
 
+        if skip_flag is None:
+            if self._update_skip_flags(module_name):
+                return
```

Comment:
Let's simplify a bit
```suggestion
        if skip_flag or (skip_flag is None and self._update_skip_flags(module_name):
            return
```

Reply:
I am not a huge fan of big `if` oneliners doing multiple checks, as those can make it harder to read and don't bring any performance benefits AFAIK.
I've merged the two nested ones, though. Let me know if that is fine for you